pymongo/asynchronous/collection.py

```diff
@@ -228,6 +228,12 @@ def __init__(
         )
         if not isinstance(name, str):
             raise TypeError("name must be an instance of str")
+        from pymongo.asynchronous.database import AsyncDatabase
+
+        if not isinstance(database, AsyncDatabase):
+            raise TypeError(
+                f"AsyncCollection requires an AsyncDatabase, {database} is an instance of {type(database)}"
```

Comment:
We should make the same `{client}` error message change to all the Database ones as well since Database's repr includes the client.